gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.java

```diff
@@ -181,8 +186,18 @@ public final void productDependency(
     public final void productDependency(@DelegatesTo(ProductDependency.class) Closure closure) {
         productDependencies.add(providerFactory.provider(() -> {
             ProductDependency dep = new ProductDependency();
-            ConfigureUtil.configureUsing(closure).execute(dep);
-            dep.isValid();
+            try {
+                ConfigureUtil.configureUsing(closure).execute(dep);
+                if (dep.getMinimumVersion() != null && dep.getMaximumVersion() == null) {
```

Comment:
This is pretty unexpected. Can we call this out in the changelog/PR description and add some docs about the new behaviour

Reply:
yep fair I can move this bit out